src/Renderer/Renderer.php

```diff
@@ -10,6 +10,8 @@
 namespace Joomla\Webservices\Renderer;
 
 use Joomla\Webservices\Application;
+use Joomla\Webservices\Renderer\RendererInterface;
+use Joomla\Webservices\Resource\Resource;
 
 use Joomla\DI\Container;
```

Comment:
So with you're much improved refactoring we don't need the container in this class anymore. We are only using the application object from the container so we should inject that directly rather than the full container into this class and all it's children.